Sources/Client/Client/Client.swift

```diff
@@ -18,16 +18,62 @@ public final class Client: Uploader {
     public typealias OnEvent = (Event) -> Void
     
     /// A client config (see `Config`).
-    public static var config = Config(apiKey: "")
-    /// A shared client.
-    public static let shared = Client()
+    @available(*, deprecated, message: """
+    Configuring the shared Client using the static `Client.config` variable has been deprecated. Please create an instance
+    of the `Client.Config` struct and call `Client.configure(_:)` to set up the shared instance of Client.
+    """)
+    public static var config: Config = .init(apiKey: "_deprecated") {
+        didSet {
+            guard _sharedClient == nil else {
+                assertionFailure("`Client.shared` instance was already used. It's not possible to change its configuration.")
+                return
+            }
+
+            configForSharedClient = config
+        }
+    }
+
+    /// The configuration object used for creating `Client.shared`.
+    private static var configForSharedClient: Config?
+
+    /// The shared client.
+    public static var shared: Client {
+        if let client = _sharedClient {
+            // Return the existing instance
+            return client
+        }
+
+        assert(
+            configForSharedClient != nil,
+            "The shared instance of the Stream chat client wasn't configured. " +
+            "Create an instance of the `Client.Config` struct and call `Client.configure(_:)` to set it up."
+        )
+
+        _sharedClient = Client(configuration: configForSharedClient ?? .init(apiKey: "_deprecated"))
+        return _sharedClient!
+    }
+
+    /// Backing variable for `Client.shared`. We need this to have finer control over its creation.
+    private static var _sharedClient: Client?
+
+    /// Configures the shared instance of `Client`.
+    ///
+    /// - Parameter configuration: The configuration object with details of how the shared instance should be set up.
+    ///
+    public static func configure(_ configuration: Config) {
+        guard _sharedClient == nil else {
+            assertionFailure("`Client.shared` instance was already used. It's not possible to change its configuration.")
+            return
+        }
+        configForSharedClient = configuration
```

Comment:
We use everywhere `config` word not `configuration`. If you want `configuration` it should be everywhere not partially.

Reply:
Good point. Changing to `config` everywhere.